### PR TITLE
Run `cargo-spellcheck` during CI

### LINKED ISSUE #1803: Check for documentation regressions with `cargo-spellcheck`
**Issue summary**

The testing infrastructure should run `cargo-spellcheck` for PRs.

**Other information and links**

### PR DESCRIPTION
**Pull request by lemmih**
**Summary of changes**
Changes introduced in this pull request:
- Only a few folders are checked at the moment. Eventually we'll just run `cargo-spellcheck` at the root level.

**Reference issue to close (if applicable)**

Closes #1803

**Other information and links**

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -125,6 +125,26 @@ jobs:
           command: audit
           args: --ignore RUSTSEC-2020-0071 --ignore RUSTSEC-2022-0040
 
+  spellcheck:
+    name: Spellcheck
+    runs-on: ubuntu-latest
+    steps:
```
Comment (LesnyRumcajs) [inline .github/workflows/rust.yml:131]: would be nice to add `Cancel previous workflows`, it takes some time to build this tool

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -125,6 +125,26 @@ jobs:
...
+      - name: Install Toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          override: true
+      - name: Rust Cache
+        uses: Swatinem/rust-cache@v1.3.0
+      - name: Install spellcheck
+        run: cargo install cargo-spellcheck
+      - name: Run Spellcheck
+        run: |
```
Comment (LesnyRumcajs) [inline .github/workflows/rust.yml:143]: Seems a bit manual, why not just `cargo spellcheck --code 1 --recursive --cfg .config/spellcheck.toml blockchain encoding forest`?

**Review by LesnyRumcajs — COMMENTED** [review]
should we also add

File: .github/workflows/rust.yml (hunk 1 above)
Comment (lemmih) [reply .github/workflows/rust.yml:131]: It should be cached, though. The latest spellcheck run took 42s in total.

File: .github/workflows/rust.yml (hunk 2 above)
Comment (lemmih) [reply .github/workflows/rust.yml:143]: I was unsure if that would work. `cargo spellcheck --recursive list-files blockchain` shows nothing, indicating that it won't traverse the sub-directories. Running `cargo spellcheck --recursive list-files blockchain/beacon` shows the expected output.

However, I tested it and it appear to work even though not all the files are shown by `list-files`.

**Review by LesnyRumcajs — APPROVED** [review]

**Comment by LesnyRumcajs** [comment]
Nice. This is the way.

**Review by elmattic — APPROVED** [review]
Good.

**Pull request merged**